src/controllers/dynakube/activegate/internal/capability/reconciler.go

```diff
@@ -44,7 +44,7 @@ func (r *Reconciler) Reconcile() error {
 		return errors.WithStack(err)
 	}
 
-	if r.dynakube.NeedsActiveGateServicePorts() {
+	if r.dynakube.NeedsActiveGateServicePorts() || r.dynakube.IsStatsdActiveGateEnabled() {
```

Comment:
Could the check for statsd be merged into the NeedsActiveGateServicePorts function?

Reply:
No, because there are some ports that are only needed, if `r.dynakube.NeedsActiveGateServicePorts()` is true and some that are only needed if `r.dynakube.IsStatsdActiveGateEnabled()` is true but for both the service has to be created

```

func CreateService(dynakube *dynatracev1beta1.DynaKube, feature string) *corev1.Service {
	var ports []corev1.ServicePort

	if dynakube.NeedsActiveGateServicePorts() {
		ports = append(ports,
			corev1.ServicePort{
				Name:       consts.HttpsServicePortName,
				Protocol:   corev1.ProtocolTCP,
				Port:       consts.HttpsServicePort,
				TargetPort: intstr.FromString(consts.HttpsServicePortName),
			},
			corev1.ServicePort{
				Name:       consts.HttpServicePortName,
				Protocol:   corev1.ProtocolTCP,
				Port:       consts.HttpServicePort,
				TargetPort: intstr.FromString(consts.HttpServicePortName),
			},
		)
	}

	if dynakube.IsStatsdActiveGateEnabled() {
		ports = append(ports,
			corev1.ServicePort{
				Name:       consts.StatsdIngestPortName,
				Protocol:   corev1.ProtocolUDP,
				Port:       consts.StatsdIngestPort,
				TargetPort: intstr.FromString(consts.StatsdIngestTargetPort),
			},
		)
	}
```

Reply:
I could create a combined property like `NeedsActiveGateService` though :thinking:

Reply:
Thats what my intention was to maybe combine those, however i think we could also leave it as it is